### PR TITLE
fix: remove check for package name and add tests

### PR DESCRIPTION
**Pull request by yuliabaron**
- [x] Ready for review
- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules
- [x] Reviewed by Snyk internal team

#### What does this PR do?
Removes init of package name when missing (for snyk test)
Adds test for package with a missing name when a lockfile is present

### CODE REVIEW

**Comment by miiila** [comment]
@yuliabaron I'm a bit confused by these changes - they bring new file into the fixture, so I assume some test should change as the behaviour should change (otherwise, why do we bring a new file to the fixtures?). 

Also, I fear there might be the case when there is a project without a lockfile and with the missing name in the `package.json`. I would appreciate one more test around this situation.

Last but not least - commit message prefix say `test`, but commit itself is touching logic, so `feat/fix/chore` is probably more suitable.

**Comment by yuliabaron** [comment]
there's one test that tests with this fixture only, and this test stubbed pkg.name with the dir name to overcome the problem I fixed (with using manifest file name for the package name if it's missing). Thus no other tests are affected. Lockfile is required because of the way we do parsing. 

when you say "commit itself is touching logic" - which change do you mean?

**Comment by Kirill89** [comment]
commit message `test: refactor tests to ES6 style` -> `refactor: tests to ES6 style`

File: test/package-no-name.test.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+const test = require('tape');
```
Comment (Kirill89) [inline test/package-no-name.test.ts]: `require` -> `import`. It's `ts` :)

File: test/acceptance/cli.acceptance.test.js [hunk 2]
```diff
@@ -1,220 +1,213 @@
...
   // We care about the request here, not the response
-  return cli.test('semver', {registry: 'npm', org: 'EFF'})
-  .then(function () {
-    var req = server.popRequest();
-    t.equal(req.method, 'GET', 'makes GET request');
-    t.match(req.url, '/vuln/npm/semver', 'gets from correct url');
-    t.equal(req.query.org, 'EFF', 'org sent as a query in request');
-  });
+  await cli.test('semver', {registry: 'npm', org: 'EFF'});
+  let req = server.popRequest();
```
Comment (Kirill89) [inline test/acceptance/cli.acceptance.test.js]: `const`

File: test/acceptance/cli.acceptance.test.js [hunk 3]
```diff
@@ -1,220 +1,213 @@
...
-  return cli.test('sinatra', {registry: 'rubygems', org: 'ACME'})
-  .then(function () {
-    var req = server.popRequest();
-    t.equal(req.method, 'GET', 'makes GET request');
-    t.match(req.url, '/vuln/rubygems/sinatra', 'gets from correct url');
-    t.equal(req.query.org, 'ACME', 'org sent as a query in request');
-  });
+test('`test sinatra --registry=rubygems` sends remote Rubygems request:', async (t) => {
+  await cli.test('sinatra', {registry: 'rubygems', org: 'ACME'});
+  let req = server.popRequest();
```
Comment (Kirill89) [inline test/acceptance/cli.acceptance.test.js]: `const`

File: test/acceptance/cli.acceptance.test.js [hunk 4]
```diff
@@ -1,220 +1,213 @@
...
+    t.match(err.message, 'Please run `bundle install`', 'shows err');
+  }
 });
 
 test('`test ruby-app --file=Gemfile.lock`', async (t) => {
   chdirWorkspaces();
   await cli.test('ruby-app', {file: 'Gemfile.lock'});
 
-  var req = server.popRequest();
+  let req = server.popRequest();
```
Comment (Kirill89) [inline test/acceptance/cli.acceptance.test.js]: `const`

File: test/acceptance/cli.acceptance.test.js [hunk 5]
```diff
@@ -230,7 +223,7 @@ test('`test ruby-app` meta when no vulns',  async (t) => {
   chdirWorkspaces();
   const res = await cli.test('ruby-app');
 
-  var meta = res.slice(res.indexOf('Organisation:')).split('\n');
+  let meta = res.slice(res.indexOf('Organisation:')).split('\n');
```
Comment (Kirill89) [inline test/acceptance/cli.acceptance.test.js]: `const`

File: test/acceptance/cli.acceptance.test.js [hunk 6]
```diff
@@ -257,7 +250,7 @@ test('`test ruby-app-thresholds`', async (t) => {
       'Tested 7 dependencies for known vulnerabilities, found 6 vulnerabilities, 7 vulnerable paths',
       '6 vulns');
 
-    var meta = res.slice(res.indexOf('Organisation:')).split('\n');
+    let meta = res.slice(res.indexOf('Organisation:')).split('\n');
```
Comment (Kirill89) [inline test/acceptance/cli.acceptance.test.js]: `const`

File: test/acceptance/cli.acceptance.test.js [hunk 7]
```diff
@@ -282,7 +275,7 @@ test('`test ruby-app-thresholds --severity-threshold=low --json`', async (t) =>
     });
     t.fail('should have thrown');
   } catch (err) {
-    var req = server.popRequest();
+    let req = server.popRequest();
```
Comment (Kirill89) [inline test/acceptance/cli.acceptance.test.js]: `const`

File: test/acceptance/cli.acceptance.test.js [hunk 8]
```diff
@@ -313,7 +306,7 @@ test('`test ruby-app-thresholds --severity-threshold=medium`', async (t) => {
...
```
Comment (Kirill89) [inline test/acceptance/cli.acceptance.test.js]: `const`

File: test/acceptance/cli.acceptance.test.js [hunk 9]
```diff
@@ -337,7 +330,7 @@ test('`test ruby-app-thresholds --severity-threshold=medium --json`', async (t)
...
```
Comment (Kirill89) [inline test/acceptance/cli.acceptance.test.js]: `const`

File: test/acceptance/cli.acceptance.test.js [hunk 10]
```diff
@@ -368,7 +361,7 @@ test('`test ruby-app-thresholds --severity-threshold=high', async (t) => {
...
```
Comment (Kirill89) [inline test/acceptance/cli.acceptance.test.js]: `const`

File: test/acceptance/cli.acceptance.test.js [hunk 11]
```diff
@@ -392,7 +385,7 @@ test('`test ruby-app-thresholds --severity-threshold=high --json`', async (t) =>
...
```
Comment (Kirill89) [inline test/acceptance/cli.acceptance.test.js]: `const`

File: test/acceptance/cli.acceptance.test.js [hunk 12]
```diff
@@ -507,10 +500,10 @@ test('`test ruby-app-no-vulns`', async (t) => {
   t.deepEqual(res, expected, '--json output is the same');
 });
 
-test('`test gradle-app` returns correct meta', function (t) {
+test('`test gradle-app` returns correct meta', async (t) => {
   chdirWorkspaces();
-  var plugin = {
-    inspect: function () {
+  let plugin = {
```
Comment (Kirill89) [inline test/acceptance/cli.acceptance.test.js]: `const`

File: test/acceptance/cli.acceptance.test.js [hunk 13]
```diff
@@ -507,10 +500,10 @@ test('`test ruby-app-no-vulns`', async (t) => {
...
+    inspect: () => {
```
Comment (Kirill89) [inline test/acceptance/cli.acceptance.test.js]: `inspect: () => {` => `inspect() {`

File: test/acceptance/cli.acceptance.test.js [hunk 14]
```diff
@@ -738,174 +724,154 @@ test('`test maven-app --file=pom.xml --dev` sends package info', async (t) => {
...
-test('`test npm-package-shrinkwrap --file=package-lock.json ` with npm-shrinkwrap errors', function (t) {
+test('`test npm-package-shrinkwrap --file=package-lock.json ` with npm-shrinkwrap errors', async (t) => {
   t.plan(1);
   chdirWorkspaces();
-  return cli.test('npm-package-shrinkwrap', {file: 'package-lock.json'})
-    .catch((e) => {
-      t.includes(e.message, '--file=package-lock.json', 'Contains enough info about error');
-    });
+  try {
+    await cli.test('npm-package-shrinkwrap', {file: 'package-lock.json'})
```
Comment (Kirill89) [inline test/acceptance/cli.acceptance.test.js]: `t.fail('expect to error');`

File: test/acceptance/cli.acceptance.test.js [hunk 15]
```diff
@@ -1595,118 +1561,111 @@ test('`test foo:latest --docker` supports custom policy', async (t) => {
...
+test('`test npm-package-with-git-url ` handles git url with patch policy', async (t) => {
   chdirWorkspaces('npm-package-with-git-url');
-  var vulns = require('./fixtures/npm-package-with-git-url/vulns.json');
+  let vulns = require('./fixtures/npm-package-with-git-url/vulns.json');
   server.setNextResponse(vulns);
-  return cli.test()
-    .catch(res => {
-      server.popRequest();
+  try {
+    await cli.test();
```
Comment (Kirill89) [inline test/acceptance/cli.acceptance.test.js]: `t.fail('expect to error');`

File: test/acceptance/cli.acceptance.test.js [hunk 16]
```diff
@@ -2309,152 +2220,152 @@ test('`protect` for unsupported package managers', function (t) {
...
-    return cli.protect({
-      'policy-path': 'custom-location',
-    })
-    .catch(function (err) {
-      var req = server.popRequest();
-      var policyString = req.body.policy;
+    try {
+      await cli.protect({
+        'policy-path': 'custom-location',
+      });
```
Comment (Kirill89) [inline test/acceptance/cli.acceptance.test.js]: `t.fail('expect to error');`

File: test/acceptance/cli.acceptance.test.js [hunk 17]
```diff
@@ -2309,152 +2220,152 @@ test('`protect` for unsupported package managers', function (t) {
...
-      cb(new Error('bail'));
+  t.test('default (insecure false)', async (t) => {
+    sinon.stub(needle, 'request').callsFake((a, b, c, d, cb) => {
+      cb(new err('bail'));
     });
     t.teardown(needle.request.restore);
-    return cli.test('npm-package')
-    .catch(function () {
+    try {
+      await cli.test('npm-package');
```
Comment (Kirill89) [inline test/acceptance/cli.acceptance.test.js]: `t.fail('expect to error');`

File: test/acceptance/cli.acceptance.test.js [hunk 18]
```diff
@@ -2309,152 +2220,152 @@ test('`protect` for unsupported package managers', function (t) {
...
     });
-    t.teardown(function () {
+    t.teardown(() => {
       delete global.ignoreUnknownCA;
       needle.request.restore();
     });
-    return cli.test('npm-package')
-    .catch(function () {
+    try {
+      await cli.test('npm-package');
```
Comment (Kirill89) [inline test/acceptance/cli.acceptance.test.js]: `t.fail('expect to error');`

**Pull request merged**